src/codewhisperer/service/featureConfigProvider.ts

```diff
@@ -0,0 +1,87 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { FeatureValue } from '../client/codewhispereruserclient'
+import { codeWhispererClient as client } from '../client/codewhisperer'
+import { AuthUtil } from '../util/authUtil'
+import { getLogger } from '../../shared/logger'
+
+export class FeatureContext {
+    constructor(public name: string, public variation: string, public value: FeatureValue) {}
+}
+
+const testFeatureName = 'testFeature'
+const featureConfigPollIntervalInMs = 30 * 60 * 1000 // 30 mins
+
+// TODO: add real feature later
+export const featureDefinitions = new Map([
+    [testFeatureName, new FeatureContext(testFeatureName, 'CONTROL', { stringValue: 'testValue' })],
+])
+
+export class FeatureConfigProvider {
+    private featureConfigs = new Map<string, FeatureContext>()
+
+    static #instance: FeatureConfigProvider
+
+    constructor() {
+        this.fetchFeatureConfigs()
+
+        setInterval(this.fetchFeatureConfigs.bind(this), featureConfigPollIntervalInMs)
+    }
+
+    public static get instance() {
+        return (this.#instance ??= new this())
+    }
+
+    async fetchFeatureConfigs(): Promise<void> {
+        if (AuthUtil.instance.isConnectionExpired()) {
+            return
+        }
+
+        getLogger().debug('Fetching feature configs')
+        try {
+            const response = await client.listFeatureEvaluations()
+
+            // Overwrite feature configs from server response
+            response.featureEvaluations.forEach(evaluation => {
+                this.featureConfigs.set(
+                    evaluation.feature,
+                    new FeatureContext(evaluation.feature, evaluation.variation, evaluation.value)
+                )
+            })
+        } catch (e) {
+            getLogger().debug('Error when fetching feature configs', e)
+        }
+        getLogger().debug(`Current feature configs: ${this.getFeatureConfigsTelemetry()}`)
+    }
+
+    getFeatureConfigsTelemetry(): string {
+        return `{${Array.from(this.featureConfigs.entries())
+            .map(([name, context]) => `${name}: ${context.variation}`)
+            .join(', ')}}`
+    }
```

Comment:
nit: add a test for this, it would be easier for ppl to generate a string to look up on ES